File: our-knowledge-front/src/backend/knowledgeService.js
import {config, appFetch} from './appFetch';

export const addKnowledge = (technologyId, technologyName, parentTechnologyId, onSuccess) =>
    appFetch('/knowledge/add', config('POST', {technologyId, technologyName, parentTechnologyId}), onSuccess);

export const deleteKnowledge = (knowledgeId, onSuccess) =>
    appFetch(`/knowledge/delete`, config('POST', {knowledgeId}), onSuccess);

export const updateKnowledge = (knowledgeId, mainSkill, likeIt, onSuccess) =>
    appFetch(`/knowledge/update`, config('POST', {knowledgeId, mainSkill, likeIt}), onSuccess);

File: OurKnowledgeBackend/src/main/java/com/example/ourknowledgebackend/service/KnowledgeService.java
package com.example.ourknowledgebackend.service;

import com.example.ourknowledgebackend.exceptions.DuplicateInstanceException;
import com.example.ourknowledgebackend.exceptions.InstanceNotFoundException;
import com.example.ourknowledgebackend.exceptions.PermissionException;
import com.example.ourknowledgebackend.model.entities.Knowledge;

import javax.naming.directory.InvalidAttributesException;
import java.util.List;

public interface KnowledgeService {

    List<Knowledge> addKnowledge(Long userId, Long technologyId, String technologyName, Long parentTechnologyId) throws InstanceNotFoundException, DuplicateInstanceException, InvalidAttributesException;

    void deleteKnowledge(Long userId, Long knowledgeId) throws InstanceNotFoundException, PermissionException;

    void updateKnowledge(Long userId, Long knowledgeId, Boolean mainSkill, Boolean likeIt) throws InstanceNotFoundException, PermissionException;
}
